zulipterminal/model.py

```diff
@@ -549,6 +549,60 @@ def send_stream_message(self, stream: str, topic: str, content: str) -> bool:
             notify_if_message_sent_outside_narrow(composition, self.controller)
         return message_was_sent
 
+    def is_unauthorised_to_post_in_stream(self, stream_id: int) -> Optional[str]:
+        """
+        Identify whether a user is authorised to send message to the given
+        stream or not using stream_post_policy and/or is_announcement_policy
+        and accordingly update the footer
+        """
+        if stream_id in self.stream_dict:
+            stream = self.stream_dict[stream_id]
+        else:
+            return "Stream ID is invalid"
+        user_info = self._all_users_by_id.get(self.user_id, None)
+        if user_info is None:
+            # Check whether the given user id is valid or not
+            return "User does not exist"
+
+        if "role" in user_info:
+            role = user_info["role"]
+        else:
+            if user_info.get("is_owner"):
+                role = 100
+            elif user_info.get("is_admin"):
+                role = 200
+            elif user_info.get("is_guest"):
+                role = 600
+            else:
+                role = 400
+
+        if stream.get("is_announcement_only") or stream.get("stream_post_policy") == 2:
+            # For admins and owners only (not using user role)
+            if role <= 200:
+                return None
+            else:
+                return STREAM_POST_POLICY[2]
+        elif stream.get("stream_post_policy") is not None:
```

Comment:
@neiljp 
In my opinion the condition can be changed to – `"stream_post_policy" in stream`
Apart from this, the rest of the function seems fine to me in the current condition.